src/Equations/Kinetic/RipplePitchScattering.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * This term implements the pitch-angle diffusion resulting when electrons
+ * interact with the magnetic ripple in a tokamak. The diffusion coefficient
+ * was first derived in [Kurzan et al, PRL 75 4626 (1995)].
+ */
+
+#include <algorithm>
+#include "DREAM/Constants.hpp"
+#include "DREAM/DREAMException.hpp"
+#include "DREAM/Equations/Kinetic/RipplePitchScattering.hpp"
+#include "DREAM/NotImplementedException.hpp"
+#include "DREAM/Settings/OptionConstants.hpp"
+#include "FVM/Interpolator1D.hpp"
+
+
+using namespace DREAM;
+using namespace std;
+
+
+/**
+ * Constructor.
+ *
+ * grid:       Kinetic grid to which this term is applied.
+ * nCoils:     Number of toroidal field coils.
+ * deltaCoils: Distance between toroidal field coils.
+ * nModes:     Number of perturbation modes given.
+ * m:          Poloidal mode
+ * n:          Toroidal mode numbers.
+ * dB_B:       Magnetic perturbation levels.
+ */
+RipplePitchScattering::RipplePitchScattering(
+    FVM::Grid *grid, enum OptionConstants::momentumgrid_type mgtype,
+    const len_t nCoils, const len_t nModes, const int_t *m,
+    const int_t *n, DREAM::IonInterpolator1D *dB_B
+) : RipplePitchScattering(
+    grid, mgtype, (2*M_PI*grid->GetRadialGrid()->GetR0() / nCoils), nModes, m, n, dB_B
+) {
+    
+    if (grid->GetRadialGrid()->GetR0() == std::numeric_limits<real_t>::infinity())
+        throw DREAMException(
+            "PitchScattermTerm: Please use the parameter 'deltaCoils' instead of "
+            "'nCoils' in cylindrical mode."
+        );
+}
+
+RipplePitchScattering::RipplePitchScattering(
+    FVM::Grid *grid, enum OptionConstants::momentumgrid_type mgtype,
+    const real_t deltaCoils, const len_t nModes, const int_t *m,
+    const int_t *n, DREAM::IonInterpolator1D *dB_B
```

Comment:
there was the question of whether we should rename IonInterpolator to MultInterpolator or similar?